src/app/core/routing/product/product.route.ts

```diff
@@ -15,7 +15,7 @@ function generateProductSlug(product: ProductView) {
     return;
   }
```

Comment:
I would rename `generateProductSlug` to `generateLocalizedProductSlug` fitting to the other namings for categories and content pages.